nomad/secure_variables_endpoint.go

```diff
@@ -468,3 +477,32 @@ func (sv *SecureVariables) authValidatePrefix(alloc *structs.Allocation, taskNam
 	}
 	return nil
 }
+
+func (sv *SecureVariables) handleCheckAndSet(path string, idx *uint64, wo structs.WriteRequest) error {
+	if idx == nil {
+		return nil
+	}
+	ci := *idx
+	var reply = new(structs.SecureVariablesReadResponse)
+	var rArgs = &structs.SecureVariablesReadRequest{
+		Path: path,
+		QueryOptions: structs.QueryOptions{
+			Region:    wo.Region,
+			AuthToken: wo.AuthToken,
+			Namespace: wo.Namespace,
+		},
+	}
+	sv.Read(rArgs, reply)
```

Comment:
Sending this back thru the top-level `Read` method means we end up having to recheck all the ACLs, set up a blocking query, and emit metrics for the `Read` RPC. We should have this read from the state store snapshot directly.